de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/cellEditor/aproperties/EReferencePropertyCellEditingSupport.java

```diff
@@ -0,0 +1,154 @@
+		adapterFactory.addAdapterFactory(new DVLMDVLMItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new DVLMStructuralItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new GeneralItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new ConceptsItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new RolesItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new UnitsItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new DVLMCategoriesItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new PropertydefinitionsItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new DVLMPropertyinstancesItemProviderAdapterFactory());
+		adapterFactory.addAdapterFactory(new ReflectiveItemProviderAdapterFactory());
+		
+		EReferencePropertyInstance propertyInstance = (EReferencePropertyInstance) caHelper.getPropertyInstance(propertyFqn);
+		EReferenceProperty propertyDefinition = ((EReferenceProperty) propertyInstance.getType());
+		
+		EReferencePropertyHelper propertyHelper = new EReferencePropertyHelper();
+		
+		Set<String> supportedFileExtensions = new HashSet<String>();
+		final EClass resolvedType = propertyHelper.getResolvedEClassType(propertyDefinition);
+		supportedFileExtensions.add(propertyHelper.getEPackageOfType(propertyDefinition).getName());
+		
+		editor = new DialogCellEditor((Composite) viewer.getControl()) {
+			
+			@Override
+			protected Object openDialogBox(Control cellEditorWindow) {
+				Object toSelect = getValue();
+				dialog = ReferenceSelectionDialog.createERefernceSelectionDialog(Display.getCurrent().getActiveShell(), resolvedType, supportedFileExtensions, adapterFactory);
+				dialog.setAllowMultiple(false);
+				dialog.setDoubleClickSelects(true);
+				setReferenceDialogInput(propertyInstance.eResource());
+				dialog.setInitialSelection(toSelect);
+				if (dialog.open() == Dialog.OK) {
+					Object selection = dialog.getFirstResult();
+					if (selection instanceof ATypeInstance) {
+						return dialog.getFirstResult();
+					}
+				} 
+				return null;
+			}
+			
+		};
+		return editor;
+	}
+	/**
+	 * An overridable method to set dialog input
+	 * @param input the input for the dialog
+	 */
+	protected void setReferenceDialogInput(Object input) {
+		dialog.setInput(input);
+	}
+	
+	@Override
+	protected Object getValue(Object element) {
+		APropertyInstance propertyInstance = getPropertyInstance(element);
+		EObject value = ((EReferencePropertyInstance) propertyInstance).getReference();
+		return value;
+	}
+	
+	@Override
+	protected boolean canEdit(Object element) {
+		boolean canEdit = createSetCommand(element, null).canExecute();
+		return canEdit;
+	}
+	
+	@Override
+	protected Command createSetCommand(Object element, Object userInputValue) {
+		APropertyInstance propertyInstance = getPropertyInstance(element);
+		if (propertyInstance != null) {

```

Comment:
Just wondering: Do we need the null check here or can the Set.Command.create deal with this case by itself?

Reply:
Not sure, I'm doing that analog to how we handle it in the ReferencePropertyCellEditingSupport

Reply:
Alright.